### ISSUE TITLE
Unexpected behavior when the search term an contains apostrophe

### ISSUE THREAD

**Issue by keith555** [typesense/typesense#666]
## Description

The dataset contains two fields, English and French. When searching using `query_by=english,french`,  behavior is expected.  When searching using `query_by=french,english` and the search term contains an apostrophe, it returns only hits where either both fields contain the search term or only the French field contains the search term. If only the English field contains the search term, there is no hit.

### Documents

```
{"english": "A dog", "french": "Un chien"}
{"english": "A horse", "french": "Un cheval"}
{"english": "A cat", "french": "Un chat"}
{"english": "A mouse", "french": "Une souris"}
{"english": "A big dog", "french": "Un gros chien"}
{"english": "A big horse", "french": "Un gros cheval"}
{"english": "A big cat", "french": "Un gros chat"}
{"english": "A big mouse", "french": "Une grande souris"}
{"english": "They can't wait", "french": "Ils ne peuvent pas attendre"}
{"english": "We can't wait", "french": "Nous can't wait"}
{"english": "Those people", "french": "can't wait"}
{"english": "He can't wait", "french": "Il ne peut pas attendre"}
{"english": "People can't wait", "french": "Les gens ne peuvent pas attendre"}
```

### Schema
```
{
"name": "'french_english'",
"fields": [
{"name": "english", "type": "string", "locale": "en" },
{"name": "french", "type": "string", "locale": "fr" }
]
}'
```

### Query
`query_by=english,french`

`http://localhost:8109/collections/english_french/documents/search?q=can't&query_by=english,french&per_page=10`

### Output
{"facet_counts":[],"found":4,"hits":[{"document":{"english":"People can't wait","french":"Les gens ne peuvent pas attendre","id":"12"},"highlights":[{"field":"english","matched_tokens":["can't"],"snippet":"People <mark>can't</mark> wait"}],"text_match":72341265420648450},{"document":{"english":"He can't wait","french":"Il ne peut pas attendre","id":"11"},"highlights":[{"field":"english","matched_tokens":["can't"],"snippet":"He <mark>can't</mark> wait"}],"text_match":72341265420648450},{"document":{"english":"We can't wait","french":"Nous can't wait","id":"9"},"highlights":[{"field":"english","matched_tokens":["can't"],"snippet":"We <mark>can't</mark> wait"}],"text_match":72341265420648450},{"document":{"english":"They can't wait","french":"Ils ne peuvent pas attendre","id":"8"},"highlights":[{"field":"english","matched_tokens":["can't"],"snippet":"They <mark>can't</mark> wait"}],"text_match":72341265420648450}],"out_of":13,"page":1,"request_params":{"collection_name":"english_french","per_page":10,"q":"can't"},"search_cutoff":false,"search_time_ms":10}

### Query 
`query_by=french,english`

`http://localhost:8109/collections/english_french/documents/search?q=can't&query_by=french,english&per_page=10`

### Output
{"facet_counts":[],"**found":2,"hits**":[{"document":{"english":"Those people","french":"can't wait","id":"10"},"highlights":[{"field":"french","matched_tokens":["can't"],"snippet":"<mark>can't</mark> wait"}],"text_match":72341265420648450},{"document":{"english":"We can't wait","french":"Nous can't wait","id":"9"},"highlights":[{"field":"french","matched_tokens":["can't"],"snippet":"Nous <mark>can't</mark> wait"}],"text_match":72341265420648450}],"out_of":13,"page":1,"request_params":{"collection_name":"english_french","per_page":10,"q":"can't"},"search_cutoff":false,"search_time_ms":17}

## Expected Behavior
Number of hits to be unaffected by `query_by` order and by use of apostrophe.

The documentation mentions that the `query_by` clause can affect the ranking, but it's not clear how it can affect the number of hits.
 Search terms without an apostrophe behave as expected.
## Actual Behavior

An apostrophe in the search term reduces the number of hits if the `query_by` clause is reversed. 

## Metadata

**Typsense Version**:
0.23.1

**OS**:
Linux ***** 4.4.0-19041-Microsoft #1237-Microsoft Sat Sep 11 14:32:00 PST 2021 x86_64 x86_64 x86_64 GNU/Linu

![Snag_195bc129](https://user-images.githubusercontent.com/1978986/180080944-cc8b3554-fc14-4e4b-b4a9-5684f218df0b.png)

**Comment by kishorenc** [comment]
Thanks for creating this issue! It's certainly an odd one -- the problem seems to be with `"locale": "fr"`. When I use `"locale": "en"` for the French field as well, the order of fields don't matter now and 5 results are produced. 

For now, if you just used `en` for both fields (or just omitted the locale property) things will work fine. However, this is certainly a strange bug, so I will dig further into this and get back to you.

**Comment by keith555** [comment]
Yes, it's very strange. I get the same error using non-English `locale`s other than `fr`, but I can work around it for the time being.  Also, I don't know it if is related, but another oddity is that searches for "o" and "v" don't return anything either.

**Comment by keith555** [comment]
Hi, this is still present in 24.1 and it's a problem where fields really need to use the locale property, for example where one field is English and another is Thai.
